docs/pages/access-controls/idps/saml-attribute-mapping.mdx

```diff
@@ -0,0 +1,198 @@
+---
+title: SAML IdP Attribute Mapping
+description: How to map user attributes to custom SAML response
+h1: SAML Attribute Mapping
+---
+
+### Prerequisites
+
+(!docs/pages/includes/no-oss-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+- User with permission to create Service Provider resource. The preset `editor` role has this permission.
+
+## Concepts
+Attribute mapping configures Teleport SAML Identity Provider to assert custom user attributes in SAML response.
+The Teleport SAML IdP supports three configurable fields for attribute mapping:
+- `name`: Name of the outgoing attribute. Required. Name should be unique across attribute mapping.
+- `value`: Value defined using a [predicate expression](../../reference/predicate-language.mdx), which can reference
+Teleport usernames, roles or traits. Required.
+- `name_format`: SAML attribute name format. Optional. The following formats are supported:
+    - `unspecified`: value equals to `urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified`. Used as a default value.
+    - `uri`: value equals to `urn:oasis:names:tc:SAML:2.0:attrname-format:uri`.
+    - `basic`: value equals to `urn:oasis:names:tc:SAML:2.0:attrname-format:basic`.
+
+Attribute mapping can be configured when adding a SAML application in Teleport web UI, or with `saml_idp_service_provider`
+resource spec created with `tctl create` or via API.
+```yaml
+kind: saml_idp_service_provider
+metadata:
+  name: example.com
+spec:
+  entity_id: https://example.com/saml/metadata
+  acs_url: https://example.com/saml/metadata
+  attribute_mapping:
+  - name: username
+    value: uid
+  - name: firstname
+    name_format: basic # optional, unspecified used as default if no value is provided.
+    value: user.spec.traits.firstname
+  - name: groups
+    name_format: urn:oasis:names:tc:SAML:2.0:attrname-format:basic # optional, full urn format.
+    value: user.spec.roles
+```
+
+## Attribute value evaluation
+
+Attribute values are authored using predicate expressions. During an SSO request, the expressions will be parsed from the
+Service Provider metadata and passed to predicate evaluator for evaluation with the authenticated user context.
+The resulting values are asserted in the SAML response under the requested attribute name.
+
+Given a correct and supported predicate expression, attributes will be mapped as long as the requested attributes are present
+in Teleport.
+Attribute mapping that points to a non-existent value will not be included in SAML assertion.
+
+## Evaluation context
```

Comment:
We should move this into the "Predicate expressions" section, otherwise it's difficult to tell what this is and how users would use it.